core/src/main/java/org/apache/accumulo/core/client/mapreduce/AccumuloMultiTableInputFormat.java

```diff
@@ -86,15 +87,8 @@ public boolean nextKeyValue() throws IOException, InterruptedException {
       }
 
       @Override
-      protected void setupIterators(TaskAttemptContext context, Scanner scanner, String tableName, RangeInputSplit split) {
-        List<IteratorSetting> iterators = split.getIterators();
-        if (null == iterators) {
-          iterators = getInputTableConfig(context, tableName).getIterators();
-        }
-
-        for (IteratorSetting setting : iterators) {
-          scanner.addScanIterator(setting);
-        }
+      protected List<IteratorSetting> contextIterators(TaskAttemptContext context, String tableName) {
+        return getInputTableConfig(context, tableName).getIterators();
```

Comment:
I notice that `mapred.MultiTableInputFormat` does not have similar logic but just uses Reader from `InputFormatBase` which appears to assume there is only one table target:
https://github.com/apache/accumulo/blob/master/core/src/main/java/org/apache/accumulo/core/client/mapred/AccumuloMultiTableInputFormat.java#L68

I am not sure if this is a bug, as it looks like, or there is some reason for this.